### DISCUSSION TITLE
Current cursor position in text box

### DISCUSSION THREAD

**Discussion by atanasj** [Hammerspoon/hammerspoon#3266] [Q&A]
I'd like to get the current cursor position in a text box, so that I can send a left or right click mouse event there. I have some code that works for left and right mouse clicks (borrowed from smarter people than me!), however will send the mouse event to the text box, but not on the spot where the text is. Is it possible to get the coordinates of the current / or last entered text? Below is my code for a left mouse click in a text box:

```lua
hs.hotkey.bindSpec(
  {"cmd", "pagedown"},
  function()
    local function getFocusedElementPosition()
      local systemElement = ax.systemWideElement()
      if not systemElement then return nil end
      local currentElement = systemElement:attributeValue("AXFocusedUIElement")
      if not currentElement then return nil end
      -- we don't want to get position for anything that isn't a text field
      if not axUtils.isTextField(currentElement) then return nil end
      local position = currentElement:attributeValue('AXPosition')
      if not position then return nil end
      return {
        x = position.x,
        y = position.y
      }
    end
    local currentApp = hs.application.frontmostApplication()
    local axApp = ax.applicationElement(currentApp)
    axApp:setAttributeValue('AXEnhancedUserInterface', true)
    axApp:setAttributeValue('AXManualAccessibility', true)
    hs.eventtap.leftClick(getFocusedElementPosition())
  end
)
```

**EDIT**

Link to code ideas for above [here](https://github.com/asmagill/hs._asm.axuielement/issues/24#issue-780507907), posted by @kittywhiskers.

**Comment by MuhammedZakir** [comment]
I found this -- https://stackoverflow.com/questions/6544311/how-to-get-global-screen-coordinates-of-currently-selected-text-via-accessibilit.

Try those `kAX...Attribute` constants mentioned in that answer. I haven't used this Hammerspoon API, but looking at your code, it looks like you have to remove `k` & `Attribute` from the name.

**Comment by atanasj** [comment]
Thanks for the tip @MuhammedZakir… I'm not a programmer, so unfortunately the skills to dissect and transplant that to `hammerspoon` are beyond me. Is there any other resources you could recommend that could help me implement your suggestion?

**Comment by muescha** [comment]
@atanasj did you find a solution?
